internal/cmd/beta/network-interface/describe/describe.go

```diff
@@ -0,0 +1,181 @@
+			ctx := context.Background()
+			model, err := parseInput(p, cmd, args)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			resp, err := req.Execute()
+			if err != nil {
+				return fmt.Errorf("describe network interface: %w", err)
+			}
+
+			return outputResult(p, model.OutputFormat, resp)
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Var(flags.UUIDFlag(), networkIdFlag, "Network ID")
+
+	err := flags.MarkFlagsRequired(cmd, networkIdFlag)
+	cobra.CheckErr(err)
+}
+
+func parseInput(p *print.Printer, cmd *cobra.Command, inputArgs []string) (*inputModel, error) {
+	nicId := inputArgs[0]
+	globalFlags := globalflags.Parse(p, cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &errors.ProjectIdError{}
+	}
+
+	model := inputModel{
+		GlobalFlagModel: globalFlags,
+		NetworkId:       flags.FlagToStringPointer(p, cmd, networkIdFlag),
+		NicId:           nicId,
+	}
+
+	if p.IsVerbosityDebug() {
+		modelStr, err := print.BuildDebugStrFromInputModel(model)
+		if err != nil {
+			p.Debug(print.ErrorLevel, "convert model to string for debugging: %v", err)
+		} else {
+			p.Debug(print.DebugLevel, "parsed input values: %s", modelStr)
+		}
+	}
+
+	return &model, nil
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient *iaas.APIClient) iaas.ApiGetNicRequest {
+	req := apiClient.GetNic(ctx, model.ProjectId, *model.NetworkId, model.NicId)
+	return req
+}
+
+func outputResult(p *print.Printer, outputFormat string, nic *iaas.NIC) error {
+	switch outputFormat {
+	case print.JSONOutputFormat:
+		details, err := json.MarshalIndent(nic, "", "  ")
+		if err != nil {
+			return fmt.Errorf("marshal network interface: %w", err)
+		}
+		p.Outputln(string(details))
+
+		return nil
+	case print.YAMLOutputFormat:
+		details, err := yaml.MarshalWithOptions(nic, yaml.IndentSequence(true))
+		if err != nil {
+			return fmt.Errorf("marshal network interface: %w", err)
+		}
+		p.Outputln(string(details))
+
+		return nil
+	default:
+		table := tables.NewTable()
+		table.AddRow("ID", *nic.Id)
+		table.AddSeparator()
+		table.AddRow("NETWORK ID", *nic.NetworkId)
+		table.AddSeparator()
+		if nic.Name != nil {
+			table.AddRow("NAME", *nic.Name)
+			table.AddSeparator()
+		}
+		if nic.Ipv4 != nil {
+			table.AddRow("IPV4", *nic.Ipv4)
+			table.AddSeparator()
+		}
+		if nic.Ipv6 != nil {
+			table.AddRow("IPV6", *nic.Ipv6)
+			table.AddSeparator()
+		}
+		table.AddRow("MAC", *nic.Mac)
+		table.AddSeparator()
+		table.AddRow("NIC SECURITY", *nic.NicSecurity)
+		if nic.AllowedAddresses != nil && len(*nic.AllowedAddresses) > 0 {
+			allowedAddreses := []string{}
+			for _, value := range *nic.AllowedAddresses {
+				allowedAddreses = append(allowedAddreses, *value.String)
+			}
```

Comment:
This should be protected by a nil check, I think the `AllowedAddresses` could be `nil`, right?

Reply:
Hmm, yeah sorry my bad, don't know what happened there